### ISSUE TITLE
$operator$GREATER_THAN_OR_EQUAL(bigint,integer):boolean not found

### ISSUE THREAD

**Issue by the6campbells** [prestodb/presto#6965]
Not sure if this is a variant of existing issues re subqueries. 
```sql
SELECT 1
   FROM "postgres_cert"."public"."tsupply"
   WHERE WIDTH_BUCKET
   (
      "qty", 0, 1000, 5
   )
   >= ( SELECT 5 FROM "postgres_cert"."public"."tsupply" WHERE "qty" = 700 )
```

```
java.sql.SQLException: Query failed (#20161224_220405_04743_ztdeg): $operator$GREATER_THAN_OR_EQUAL(bigint,integer):boolean not found
```

**Comment by kokosing** [comment]
can you post `EXPLAIN` output for this query as well as schema of `"postgres_cert"."public"."tsupply"`?

**Comment by the6campbells** [comment]
Not using an actual  table will repro the issue 

   SELECT 1
   FROM ( values (1) ) T1 (qty)
   WHERE WIDTH_BUCKET
   (
      "qty", 0, 1000, 5
   )
   >= ( SELECT c1 from ( values (5) ) T2 (c1) )

- Output[_col0] => [expr_7:integer]
        _col0 := expr_7
    - Project => [expr_7:integer]
            expr_7 := 1
        - InnerJoin[("width_bucket"(CAST("field" AS double), 0.0, 1000.0, BIGINT '5') >= "field_1")] => [field:integer, field_1:integer]
            - LocalExchange[ROUND_ROBIN] () => field:integer
                - Values => [field:integer]
                        (1)
            - Scalar => [field_1:integer]
                - Values => [field_1:integer]
                        (5)


Original scenario

- Output[_col0] => [expr_8:integer]
        _col0 := expr_8
    - RemoteExchange[GATHER] => expr_8:integer
        - Project => [expr_8:integer]
                expr_8 := 1
            - InnerJoin[("width_bucket"(CAST("qty" AS double), 0.0, 1000.0, BIGINT '5') >= "expr")] => [qty:integer, expr:integer]
                - TableScan[postgres_cert:postgres_cert:public.tsupply:null:public:tsupply, originalConstraint = true] => [qty:integer]
                        qty := JdbcColumnHandle{connectorId=postgres_cert, columnName=qty, columnType=integer}
                - RemoteExchange[REPLICATE] => expr:integer
                    - Scalar => [expr:integer]
                        - RemoteExchange[GATHER] => expr:integer
                            - Project => [expr:integer]
                                    expr := 5
                                - Filter[("qty_4" = 700)] => [qty_4:integer]
                                    - TableScan[postgres_cert:postgres_cert:public.tsupply:null:public:tsupply, originalConstraint = ("qty_4" = 700)] => [qty_4:integer]
                                            qty_4 := JdbcColumnHandle{connectorId=postgres_cert, columnName=qty, columnType=integer}



CREATE TABLE "public"."tsupply"
(
   rnum int NOT NULL,
   sno varchar(2),
   pno varchar(2),
   jno varchar(2),
   qty int
)
;

**Comment by kokosing** [comment]
I tested that on current master and it works. What version do you use?

```
presto:tiny> SELECT 1 FROM ( values (1) ) T1 (qty) WHERE WIDTH_BUCKET ( "qty", 0, 1000, 5 ) >= ( SELECT c1 from ( values (5) ) T2 (c1) );
 _col0 
-------
(0 rows)

Query 20170103_080513_00005_z5rkq, FINISHED, 1 node
Splits: 35 total, 35 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

presto:tiny> explain SELECT 1 FROM ( values (1) ) T1 (qty) WHERE WIDTH_BUCKET ( "qty", 0, 1000, 5 ) >= ( SELECT c1 from ( values (5) ) T2 (c1) );
                                                              Query Plan                                                              
--------------------------------------------------------------------------------------------------------------------------------------
 - Output[_col0] => [expr_8:integer]                                                                                                  
         _col0 := expr_8                                                                                                              
     - Project => [expr_8:integer]                                                                                                    
             expr_8 := 1                                                                                                              
         - InnerJoin[("width_bucket"(CAST("field" AS double), 0.0, 1000.0, BIGINT '5') >= "expr_7")] => [field:integer, expr_7:bigint]
             - LocalExchange[ROUND_ROBIN] () => field:integer                                                                         
                 - Values => [field:integer]                                                                                          
                         (1)                                                                                                          
             - LocalExchange[SINGLE] () => expr_7:bigint                                                                              
                 - Project => [expr_7:bigint]                                                                                         
                         expr_7 := CAST("field_1" AS bigint)                                                                          
                     - LocalExchange[ROUND_ROBIN] () => field_1:integer                                                               
                         - Scalar => [field_1:integer]                                                                                
                             - Values => [field_1:integer]                                                                            
                                     (5)                                                                                              
                                                                                                                                      
(1 row)
```

**Comment by the6campbells** [comment]
Presto 0.160

**Comment by kokosing** [comment]
So it should be fixed in 0.162 via https://github.com/prestodb/presto/commit/0d5699dcefefd5e21c0aaf2a9b1f6b1555cad647.

I am closing this, please reopen if it will occur in 0.162.
